Lib/pathlib.py

```diff
@@ -673,7 +676,11 @@ def is_absolute(self):
         # ntpath.isabs() is defective - see GH-44626 .
         if self._flavour is ntpath:
             return bool(self.drive and self.root)
-        return self._flavour.isabs(self._raw_path)
+        else:
+            for path in self._raw_paths:
```

Comment:
To check I understand correctly: the rewrite of this method relies on am invariant that if any of the segments in an unnormalised path is absolute, the path as a whole will be absolute. Is that correct? If so, do we have assertions anywhere that test this invariant?

Reply:
The invariant is true on POSIX, but not necessarily other platforms. This code isn't ideal! I ought to handle the POSIX case explicitly.

Reply:
Thanks for the great feedback btw. I've rewritten the method to be a little more explicit - what do you think?

Reply:
Thanks, that's much clearer now!